components/Roster/RosterList.vue

```diff
@@ -25,48 +27,20 @@
       selectedJid: String,
     },
 
-    data() {
-      return {
-        selectedSection: -1,
-        selectedItem: -1,
-      }
-    },
-
     computed: {
-      jidLocations() {
-        const locations = new Map();
-
-        for (let i = 0; i < this.items?.length || 0; i++) {
-          for (let j = 0; j < this.items[i].items?.length || 0; j++) {
-            locations.set(this.$stanza.toBare(this.items[i].items[j].jid), [i, j]);
-          }
-        }
-
-        return locations;
-      },
-    },
+      selectedSectionsItems() {
```

Comment:
The previous 'selected JID' system was a mess, and didn't support multiple selected items, either.

The new system stores selections as -- more-or-less -- `Map<Index, [Index]>`. Then, each `<RosterSection/>` gets fed its own `[Index]` value, and then, finally, each item is checked if its index is in the selection array.

The algorithm is just:
```hs
map snd . filter ((== jid) . :jid) . zip [0..] <$> items
```